shark_turbine/kernel/wave/codegen.py

```diff
@@ -293,24 +256,11 @@ def handle_add(emitter: WaveEmitter, node: fx.Node):
     lhs = cast_py_value(emitter, lhs)
     rhs = cast_py_value(emitter, rhs)
 
-    def get_type(value: IRProxyValue):
-        return value.ir_value.type
+    if lhs.ir_value.type != rhs.ir_value.type:
+        raise ValidationError("Expected lhs and rhs to have same type.")
 
-    # TODO: this is too hard coded
-    if (lhs_type := get_type(lhs)) != (rhs_type := get_type(rhs)):
-        if isinstance(lhs_type.element_type, F32Type):
-            lhs = arith_d.truncf(
-                VectorType.get(lhs_type.shape, F16Type.get()), lhs.ir_value
-            )
-            rhs = rhs.ir_value
-        elif isinstance(rhs_type.element_type, F32Type):
-            lhs = lhs.ir_value
-            rhs = arith_d.truncf(
-                VectorType.get(rhs_type.shape, F16Type.get()), rhs.ir_value
-            )
-    else:
-        lhs = lhs.ir_value
-        rhs = rhs.ir_value
+    lhs = lhs.ir_value
+    rhs = rhs.ir_value
     result = arith_d.addf(lhs, rhs)
```

Comment:
How about support for addi? I think those are the only 2 possibilities?

Reply:
Added, good catch, thanks!

Reply:
thanks!